pkg/apis/v1beta1/nodeclass_validation.go

```diff
@@ -112,10 +119,16 @@ func (in *NodeClassSpec) validateAMISelectorTerms() (errs *apis.FieldError) {
 	return errs
 }
 
+//nolint:gocyclo
 func (in *AMISelectorTerm) validate() (errs *apis.FieldError) {
 	errs = errs.Also(validateTags(in.Tags).ViaField("tags"))
 	if len(in.Tags) == 0 && in.ID == "" && in.Name == "" && in.SSM == "" {
 		errs = errs.Also(apis.ErrGeneric("expect at least one, got none", "tags", "id", "name", "ssm"))
+	} else if len(in.Tags) > 0 && in.ID != "" ||
+		in.Name != "" && in.ID != "" ||
+		in.SSM != "" && in.ID != "" ||
+		in.Owner != "" && in.ID != "" {
```

Comment:
I think this would look cleaner:
```suggestion
	} else if in.ID != "" && (len(in.Tags) > 0 ||
		in.Name != "" ||
		in.SSM != "" ||
		in.Owner != "") {
```

Reply:
Good call 👍